### PR TITLE
dnsdist: Ignore TCAction over TCP

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Supersedes #10693.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/docs/rules-actions.rst [hunk 1]
```diff
@@ -47,10 +47,13 @@ If this is not enough, try::
 
 or::
 
-  addAction(MaxQPSIPRule(5), TCAction())
+  addAction(AndRule{MaxQPSIPRule(5), TCPRule(false)}, TCAction())
 
 This will respectively drop traffic exceeding that 5 QPS limit per IP or range, or return it with TC=1, forcing clients to fall back to TCP.
 
+In that last one, note the use of :func:`TCPRule`.
+Without it, clients would get TC=1 even if they correctly fell back to TCP.
```
Comment (Habbie) [inline pdns/dnsdistdist/docs/rules-actions.rst:55]: Maybe note the 1.7.0 change here too?

**Review by Habbie — APPROVED** [review]
LGTM, with one doc suggestion.

File: pdns/dnsdistdist/docs/rules-actions.rst (hunk 1 above)
Comment (rgacogne) [reply pdns/dnsdistdist/docs/rules-actions.rst:55]: I pondered that but could not find a way not involving many words, and figured that check was cheap anyway :-)

Comment (Habbie) [reply pdns/dnsdistdist/docs/rules-actions.rst:55]: Fair!

**Pull request merged**